ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -19,112 +17,84 @@ export interface DropDownState {
-                fromEvent(document, 'click').pipe(
-                    filter((event: Event) => {
-                        return this.content && this.state.opened && !this.content.contains(event.target as Node) && !this.el.contains(event.target as Node);
-                    })
-                )
-            ).subscribe(() => {
-                this.close();
-            }),
-            fromEvent(document, 'scroll', {capture: true}).subscribe(() => {
-                if (this.state.opened && this.content && this.el) {
-                    this.setState(this.refreshState());
-                }
-            })
-        ];
-    }
-
-    public componentWillUnmount() {
-        (this.subscriptions || []).forEach(s => s.unsubscribe());
-        this.subscriptions = [];
-    }
-
-    public close() {
-        this.setState({opened: false});
-    }
-
-    private refreshState() {
-        const anchor = this.el.querySelector('.argo-dropdown__anchor') as HTMLElement;
-        const {top, left} = anchor.getBoundingClientRect();
-        const anchorHeight = anchor.offsetHeight + 2;
-
-        const newState = {left: this.state.left, top: this.state.top, opened: this.state.opened};
         // Set top position
-        if (this.content.offsetHeight + top + anchorHeight > window.innerHeight) {
-            newState.top = top - this.content.offsetHeight - 2;
+        if (contentEl.offsetHeight + anchorTop + anchorHeight > window.innerHeight) {
+            newState.top = anchorTop - contentEl.offsetHeight - 2;
         } else {
-            newState.top = top + anchorHeight;
+            newState.top = anchorTop + anchorHeight;
         }
 
         // Set left position
-        if (this.content.offsetWidth + left > window.innerWidth) {
-            newState.left = left - this.content.offsetWidth + anchor.offsetWidth;
+        if (contentEl.offsetWidth + anchorLeft > window.innerWidth) {
+            newState.left = anchorLeft - contentEl.offsetWidth + anchorEl.offsetWidth;
         } else {
-            newState.left = left;
+            newState.left = anchorLeft;
         }
+
         return newState;
     }
 
-    private open() {
-        if (!this.content || !this.el) {
+    function open() {
+        if (!contentRef || !anchorRef) {
             return;
         }
 
-        const newState = this.refreshState();
+        const newState = refreshState();
+
         newState.opened = true;
-        this.setState(newState);
-        dropDownOpened.next(this);
+        setOpened(newState.opened);
+        setLeft(newState.left);
+        setTop(newState.top);
+    }
+
+    function close(event: MouseEvent) {
+        if (opened) {
+            // Doesn't close when clicked inside the portal area
+            if (contentEl.contains(event.target as Node) || anchorEl.contains(event.target as Node)) {
+                return;
+            }
+
+            setOpened(false);
+        }
```

Comment:
```suggestion
        if (!opened) {
            return
        }
        
        // don't close when clicked inside the portal area
        if (contentEl.contains(event.target as Node) || anchorEl.contains(event.target as Node)) {
            return;
        }

        setOpened(false);
```

we can invert the conditional to simplify this a bit and reduce nesting

Reply:
actually, with my further comments, I think we can remove the `!opened` check from here as it would be in the effect instead.